tests/wpt/mozilla/tests/mozilla/worklets/test_paint_worklet_timeout.js

```diff
@@ -0,0 +1,5 @@
+registerPaint("testgreen", class {
+    paint(ctx, size) {
+        for(;;); // take a bit too long to do initialization
```

Comment:
It would be nice if there was a blocking API we could use here rather than busy-waiting.

Reply:
Hey, I tried playing with a _synchronous_ XHR to `/`, but that doesn't seem to trigger the timeout (the request still returns in time for the timeout). Do you have any other suggestions for such an API? I've pushed changes to cover the rest of your suggestions!

Reply:
I think the XHR API isn't available to worklets unfortunately. Can you check to see if you're getting an error from XHR being undefined or if the HTTP request is going through. If it's the latter, there may be some wpt cleverness for getting the HTTP server to delay responding.

Reply:
XHR should not even be defined in worklet globals; if it is, that's an issue that should be filed :)

Reply:
@jdm ack, indeed it seems it isn't available. For some reason I am not seeing error messages when I run the debug profile (but can see log statements). However, my paint() function doesn't execute any subsequent log statements after I use `window.XMLHttpRequest` or `XMLHttpRequest`.

Any other ideas for a slow/blocking API?

Reply:
If we had the `Atomics` API then we could use `wait`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics/wait but I don't think we do :/ I think we're stuck with a busy-wait. @jdm, should we check in the busy-wait, even though it will probably thrash one core of the test machine until the wpt test times out?

Reply:
Is there a test-only API we could implement that would do what we want?

Reply:
We could implement something like synchronous sleep, and hide it behind a pref.

Reply:
That's a good idea. I can also try implementing this, if you show me the pieces of a similar existing API.

Reply:
There's a test-only worklet API in https://github.com/servo/servo/blob/master/components/script/dom/webidls/TestWorkletGlobalScope.webidl, and a test-only iterable interface in https://github.com/servo/servo/blob/master/components/script/dom/webidls/TestBindingIterable.webidl. These should be good models to follow.